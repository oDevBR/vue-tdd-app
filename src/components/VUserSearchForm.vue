
<script>
export default {
  name: 'UserSearchForm',

  data() {
    return {
      username: ''
    }
  }
}
</script>

<template>
  <form @submit.prevent="$emit('submitted', username)" style="display: flex; justify-content: center;">
    <el-input v-model="username" class="search-form__input" placeholder="Pesquise o usuÃ¡rio" style="width: 50vw;">
      <el-button class="search-form__button" native-type="submit" slot="append" icon="el-icon-search"></el-button>
    </el-input>
  </form>
</template>
