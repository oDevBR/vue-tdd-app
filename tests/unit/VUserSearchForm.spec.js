import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
import Element from 'element-ui'
import VUserSearchForm from '@/components/VUserSearchForm'

const localVue = createLocalVue()
localVue.use(Element)

describe('VUserSearchForm', () => {
    const build = () => {
        const options = { localVue }
        const wrapper = shallowMount(VUserSearchForm, options)
        const wrapperMounted = mount(VUserSearchForm, options)
        return {
            wrapper,
            wrapperMounted,
            input: () => wrapper.find('.search-form__input'),
            inputMounted: () => wrapperMounted.find('input'),
            button: () => wrapperMounted.find('button')
        }
    }

    it('should renders the component', () => {
        const { wrapper } = build()
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should renders main child components', () => {
        const { input, button } = build()

        expect(input().exists()).toBe(true)
        expect(button().exists()).toBe(true)
    })

    it('should calls "submitted" event when submitting form', () => {
        const expectedUser = 'whisoer'
        const { wrapperMounted, inputMounted, button } = build()

        inputMounted().element.value = expectedUser
        inputMounted().trigger('input')

        button().trigger('click')
        button().trigger('submit')
    
        expect(wrapperMounted.emitted().submitted[0]).toEqual([expectedUser])
    })
})