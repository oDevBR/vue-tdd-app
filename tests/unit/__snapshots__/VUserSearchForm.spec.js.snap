// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUserSearchForm should renders the component 1`] = `
<form style="display: flex; justify-content: center;">
  <el-input-stub value="" type="text" autocomplete="off" validateevent="true" placeholder="Pesquise o usuÃ¡rio" class="search-form__input" style="width: 50vw;">
    <el-button-stub type="default" icon="el-icon-search" nativetype="submit" class="search-form__button"></el-button-stub>
  </el-input-stub>
</form>
`;
